### ISSUE TITLE
[no-undef]: add option to ignore identifiers/var

### ISSUE THREAD

**Issue by maddhruv** [eslint/eslint#14088]
**What rule do you want to change?**

`no-undef`

**Does this change cause the rule to produce more or fewer warnings?**

`fewer warnings`

**How will the change be implemented? (New option, new default behavior, etc.)?**

New option to be added

```js
{
    ignore: {
        type: 'array',
        default: []
    }
}
```

**Please provide some example code that this change will affect:**

To certain variables and features that are globally defined and are language scoped

```js
export type Foo = $Values<typeof Bar>;
```

**What does the rule currently do for this code?**

`$Values is not defined`

**What will the rule do after it's changed?**

Won't through an error here

**Are you willing to submit a pull request to implement this change?**

Yes

**Comment by mdjermanovic** [comment]
Hi @maddhruv, thanks for the issue!

If you specify `$Values` as a [global](https://eslint.org/docs/user-guide/configuring/language-options#specifying-globals) in your ESLint configuration, ESLint will treat it as a globally defined variable and `no-undef` won't report it.

Or, you can install [eslint-plugin-flowtype](https://github.com/gajus/eslint-plugin-flowtype), and add `"extends": ["plugin:flowtype/recommended"]` to your configuration or just enable `flowtype/define-flow-type` rule.

**Comment by maddhruv** [comment]
Hey @mdjermanovic I know that this can be solved with `eslint-plugin-flowtype`, but this is just an example of a case where one would want to ignore an undef variable. Also, for some scenarios, one may not want to declare it as global.

**Comment by mdjermanovic** [comment]
I'm not in favor of this change as it encourages ignoring broken references instead of declaring globals in ESLint configuration. Correct resolution of variables is important for a number of ESLint rules, so it should be always better to declare a variable than to have unresolved references in the source code.

If something shouldn't be actually a reference to a variable, then the scope analysis (provided by parsers) shouldn't treat it as such.

**Comment by nzakas** [comment]
I’m also not in favor of this change for the same reasons. There are numerous workarounds already available, including using a disable comment in that location.

**Comment by maddhruv** [comment]
Okay! Thanks for the responses. Closing
